include/SFML/System/Vector2.hpp

```diff
@@ -247,6 +247,55 @@ template <typename T>
 template <typename T>
 [[nodiscard]] constexpr bool operator !=(const Vector2<T>& left, const Vector2<T>& right);
 
+//Esteban Calvo start here
+//Enhancements
+//Dot product of 2 vectors
+
+////////////////////////////////////////////////////////////
+/// \relates Vector2
+/// \brief Overload of binary operator *
+///
+/// This operator performs the cross product of two vectors,
+///
+/// \param left  Left operand (a vector)
+/// \param right Right operand (a vector)
+///
+/// \return value of cross product of type T
+///
+////////////////////////////////////////////////////////////
+template<typename T>
+[[nodiscard]] constexpr T operator*(Vector2<T>& left, Vector2<T>& right);
+
+////////////////////////////////////////////////////////////
+/// \relates Vector2
+/// \brief calculates cross product of a vector
+///
+/// This operator performs the cross product of two vectors,
+///
+/// \param left  Left operand (a vector)
+/// \param right Right operand (a vector)
+///
+/// \return value of cross product of type T. For 2D array, cross product will give z value
+///
+////////////////////////////////////////////////////////////
+template<typename T>
+[[nodiscard]] constexpr T cross(Vector2<T>& left, Vector2<T>& right);
```

Comment:
The parameters should be const references, not mutable references since we want to avoid an extra copy but also want to allow temporary objects to be passed to this function.